### PR TITLE
Updated netspeed widget

### LINKED ISSUE #66: Improve netspeed widget icons
- [ ] Use a reliable way to detect interface type (wifi, ethernet, etc)
- [ ] Detect when network is down and show icon accordingly
- [ ] Add a config flag to show local IP address (maybe public?)

This is the proposed design for down network

image

image

Icons:
- Wifi: `f16c2`
- Ethernet: `f0319`

Proposed design por IP address
 
image

Icon `f0a5f`

### PR DESCRIPTION
**Pull request by Stealthii**
This PR implements the features suggested in #66:
- Reliably detect wireless/wired interface type
- Detect when the interface is down
- Add config flag to show local IPv4 address
- Updated icons for interface types and state

The following additional changes were implemented:
- Reduce to single bc calls for netspeed calculations
- Add interface auto-detection via route
- Add variable refresh time

BATS tests have been added to validate functionality, with a few edge cases handled that were discovered during testing.

---

## Minor changes

- [X] netspeed: auto-detect interface if undefined
- [X] netspeed: new widget icons for device class and state
- [X] netspeed: detect interface type (wired/wireless)
- [X] netspeed: detect interface online state
- [X] netspeed: show IPv4 address with the `@tokyo-night-tmux_netspeed_showip` option
- [X] netspeed: vary the refresh rate with the `@tokyo-night-tmux_netspeed_refresh` option

## Bug & security fixes

- [X] netspeed: improve accuracy and performance of netspeed calculations with `bc`
- [X] netspeed: handle 
- [X] netspeed: handle VPN devices on macOS using the primary interface

## Additional information

Potentially resolves #66.

### CODE REVIEW

**Comment by Stealthii** [comment]
Bats tests may be run as part of the PR workflow if #70 gets implemented.

File: src/wb-git-status.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ REVIEW_STATUS=""
 ISSUE_STATUS=""
 BUG_STATUS=""
 
-if [[ -z "$BRANCH" ]]; then
+if [[ -z $BRANCH ]]; then
```
Comment (janoamaral) [inline src/wb-git-status.sh:27]: This is simple and we can just overlook, but try to avoid changes outside the scope of the issue

**Review by janoamaral — CHANGES_REQUESTED** [review]
Just change the icons and we are good

File: src/netspeed.sh [hunk 2]
```diff
@@ -4,70 +4,71 @@
...
-# Convert into readable format
-readable_format() {
-  local bytes=$1
+# Icons
+declare -A NET_ICONS
+NET_ICONS[wifi_up]="#[fg=${THEME[foreground]}]\U000f05a9"  # nf-md-wifi
+NET_ICONS[wifi_down]="#[fg=${THEME[red]}]\U000f05aa"       # nf-md-wifi_off
+NET_ICONS[wired_up]="#[fg=${THEME[foreground]}]\U000f0318" # nf-md-lan_connect
+NET_ICONS[wired_down]="#[fg=${THEME[red]}]\U000f0319"      # nf-md-lan_disconnect
+NET_ICONS[traffic_tx]="#[fg=${THEME[bblue]}]\U000f0cd8"    # nf-md-upload_network_outline
```
[the remark this answers is not in the document]
Comment (Stealthii) [reply src/netspeed.sh]: Fixed in ec96f51.

File: src/netspeed.sh [hunk 3]
```diff
@@ -4,70 +4,71 @@
...
+NET_ICONS[traffic_rx]="#[fg=${THEME[bgreen]}]\U000f0c66"   # nf-md-download_network_outline
```
[the remark this answers is not in the document]
Comment (Stealthii) [reply src/netspeed.sh]: Fixed in ec96f51.

File: src/wb-git-status.sh (hunk 1 above)
Comment (Stealthii) [reply src/wb-git-status.sh]: Understood. That linting fix was applied in the BATS PR, and has been removed from this PR after rebasing.

**Comment by Stealthii** [comment]
PR rebased, icons and screenshot updated!

**Review by janoamaral — APPROVED** [review]
